### PR TITLE
AO3-4779 Invitations strong params

### PR DESCRIPTION
**Pull request by mattsears**
# Pull Request Checklist

* [ x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [x] Have you added tests for any changed functionality?
* [x] Have you added the JIRA issue number as the *first* thing in your pull request title (eg: `AO3-1234 Fix thing`)
* [x] Have you updated the JIRA issue with the information below?

## Issue

https://otwarchive.atlassian.net/browse/AO3-4779

## Purpose

Implements strong parameters and Invitations.

## Testing

1. Log in as testy
2. Hi, username! > My Dashboard > Invitations > Manage Invitations
3. Click on the token for an invitation that has a link in it's copy and use column to go to the invitation's page, e.g. http://test.archiveofourown.org/users/testy/invitations/355584
4. Edit the address in the "Enter an email address" field
5. Press "Update Invitation"

You should be able to change this without any unexpected error messages.
You should also do roughly the same thing as an admin:
1. Log in as admin
2. Invitations > Invite New Users
3. Under "Track Invitations," enter "testy" in the "Enter a user name" field and press "Go" to see a list of testy's invitations
4. Click on the token for an invitation that has a link in it's copy and use column to go to the invitation's page, e.g. http://test.archiveofourown.org/users/testy/invitations/355584
5. Edit the address in the "Enter an email address" field
6. Press "Update Invitation"

## References

None

## Credit

Littlelines - Matt Sears

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?*

he/him

### CODE REVIEW

File: app/views/invitations/index.html.erb [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <!--Descriptive page name, messages and instructions-->
```
Comment (sarken) [inline app/views/invitations/index.html.erb]: I think [/views/invitations/_invitation.html.erb](https://github.com/otwcode/otwarchive/blob/master/app/views/invitations/_invitation.html.erb) might also need to be updated -- I got `Required parameter missing: invitation` when I tried to use the email field on my dev environment.

Comment (mattsears) [inline app/views/invitations/index.html.erb]: I'm having a little trouble reproducing this one. The form wrappers invitation i.e `form_for(invitation)` so it should be submitting the `invitation` parameter. Could you possible list the steps to reproduce this?

Comment (sarken) [inline app/views/invitations/index.html.erb]: Sure!

1. Log in as admin
2. In the "Invitations" menu, choose "Invite New Users"
3. Under "Track invitations," enter a username , e.g. testy, and press "Go"
4. Click the name "testy" to go to testy's "Create more invitations for this user" page
5. Fill in "Number of invitations" with some number and press "Create"
6. Repeat steps 2-3 to get updated search results for testy's invitations
7. Click the token for one of the invitations you just gave testy -- that will take you to an ugly, detailed view of the invitation
8. Fill in "Enter an email address" with an email address and press "Update Invitation"

The important thing is "Enter an email address" needs to have been blank to start with -- if there was already an address in there, it works fine. Also, weirdly, the email information does seem to save even if you get the `Required parameter missing: invitation` error. 

You can add this test to bottom of features/admins/admin_invitations.feature if you'd like: 
```
  Scenario: An admin can edit an invitation
    Given the user "dax" exists and is activated
      And "dax" has "2" invitations
      And I am logged in as an admin
    When I follow "Invite New Users"
      And I fill in "Enter a user name" with "dax"
      And I press "Go"
    Then I should see "copy and use"
    When I follow "Invite New Users"
      And I fill in "Enter an invite token" with "dax's" invite code
      And I press "Go"
    Then I should see "copy and use"
    When I fill in "invitation_invitee_email" with "oldman@ds9.com"
      And I press "Update Invitation"
    Then I should see "oldman@ds9.com" in the "invitation_invitee_email" input
```

When I run it, the failure doesn't look like the strong parameters errors we see on Travis, but it does fail on the last step, and the HTML screenshot it creates is a white page with a `Required parameter missing: invitation` error. (It does pass on master.)

Comment (mattsears) [inline app/views/invitations/index.html.erb]: Thanks!!! That was super helpful! I've pushed a few updates that should resolve the issue.

Comment (sarken) [inline app/views/invitations/index.html.erb]: I think this is ready, then -- thanks!

**Pull request merged**